src/pdf_beamtime/src/pdf_beamtime_server.cpp

```diff
@@ -51,13 +52,32 @@ PdfBeamtimeServer::PdfBeamtimeServer(
   current_state_ = State::HOME;
   gripper_present_ = node_->get_parameter("gripper_present").as_bool();
   inner_state_machine_ = new InnerStateMachine(node_);
+
+  bluesky_interrupt_service_ = interrupt_node_->create_service<BlueskyInterruptMsg>(
+    "bluesky_interrupt",
+    std::bind(
+      &PdfBeamtimeServer::bluesky_interrupt_cb, this, _1, _2));
+}
+
+void PdfBeamtimeServer::bluesky_interrupt_cb(
+  const std::shared_ptr<BlueskyInterruptMsg::Request> request,
+  std::shared_ptr<BlueskyInterruptMsg::Response> response)
+{
+  handle_pause();
 }
 rclcpp::node_interfaces::NodeBaseInterface::SharedPtr PdfBeamtimeServer::getNodeBaseInterface()
 // Expose the node base interface so that the node can be added to a component manager.
 {
   return node_->get_node_base_interface();
 }
 
+rclcpp::node_interfaces::NodeBaseInterface::SharedPtr PdfBeamtimeServer::getNodeBaseInterface_BIN()
```

Comment:
Don't mix camel case and underscores. 
I'd refactor to  `getInterruptNodeBaseInterface`. Or whatever interrupt get's changed to.

Reply:
Function name is changed to `getInterruptNodeBaseInterface`.
https://github.com/maffettone/erobs/blob/b7595c9042764872d4c9c7d38f3960a33446b96c/src/pdf_beamtime/src/pdf_beamtime_server.cpp#L85